spring-cloud-stream/src/main/java/org/springframework/cloud/stream/function/IntegrationFlowFunctionSupport.java

```diff
@@ -0,0 +1,138 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.function;
+
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import org.reactivestreams.Publisher;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import org.springframework.cloud.function.context.FunctionCatalog;
+import org.springframework.cloud.function.context.catalog.FunctionInspector;
+import org.springframework.cloud.function.core.FluxSupplier;
+import org.springframework.cloud.stream.converter.CompositeMessageConverterFactory;
+import org.springframework.integration.dsl.IntegrationFlowBuilder;
+import org.springframework.integration.dsl.IntegrationFlows;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+
+/**
+ *
+ * @author Oleg Zhurakousky
+ *
+ * @since 2.1
+ */
+public class IntegrationFlowFunctionSupport {
+
+	private final FunctionCatalog functionCatalog;
+
+	private final FunctionInspector functionInspector;
+
+	private final CompositeMessageConverterFactory messageConverterFactory;
+
+	private final FunctionProperties functionProperties;
+
+	/**
+	 *
+	 * @param functionCatalog
+	 * @param functionInspector
+	 * @param messageConverterFactory
+	 * @param functionProperties
+	 */
+	public IntegrationFlowFunctionSupport(FunctionCatalog functionCatalog, FunctionInspector functionInspector,
+			CompositeMessageConverterFactory messageConverterFactory, FunctionProperties functionProperties) {
+
+		Assert.notNull(functionCatalog, "'functionCatalog' must not be null");
+		Assert.notNull(functionInspector, "'functionInspector' must not be null");
+		Assert.notNull(messageConverterFactory, "'messageConverterFactory' must not be null");
+		this.functionCatalog = functionCatalog;
+		this.functionInspector = functionInspector;
+		this.messageConverterFactory = messageConverterFactory;
+		this.functionProperties = functionProperties;
+	}
+
+	/**
+	 * Creates an instance of the {@link IntegrationFlowBuilder} from a {@link Supplier} bean available in the context.
+	 * The name of the bean must be provided via `spring.cloud.stream.function.name` property.
+	 *
+	 * @return instance of {@link IntegrationFlowBuilder}
+	 */
+	public <O> IntegrationFlowBuilder integrationFlowFromNamedSupplier() {
+		if (StringUtils.hasText(this.functionProperties.getName())) {
+			Supplier<?> supplier = functionCatalog.lookup(Supplier.class, this.functionProperties.getName());
+			if (supplier instanceof FluxSupplier) {
+				supplier = ((FluxSupplier<?>)supplier).getTarget();
+			}
+			return this.integrationFlowFromProvidedSupplier(supplier);
+		}
+		throw new IllegalStateException("Supplier name is not provided");
```

Comment:
I think this message should include an appropriate configuration property to lead people to the correct direction.
That always was a bonus from Spring: not only say what is wrong, but also point to the right direction to fix.